Sources/BSON/Document.swift

```diff
@@ -1,77 +1,184 @@
 import Foundation
 import NIO
 
+/// This shared allocator instance should be used for all underlying `ByteBuffer` creation.
+private let BSON_ALLOCATOR = ByteBufferAllocator()
+/// Maximum BSON document size in bytes
+private let BSON_MAX_SIZE = 0x100000
```

Comment:
is this actually the max size? I thought the [max size](https://docs.mongodb.com/manual/reference/limits/#BSON-Document-Size) was 16 MB or 16000000 bytes.